### PR TITLE
Add set_agent_group and set_agent_group_context UT

### PR DESCRIPTION
**Pull request by palaciosjeremias**
|Related issue|
|---|
|#11754|

## Description
This PR adds UT for:
- **wdb_global_set_agent_group_context()**
- **wdb_global_set_agent_groups()**

It also fixes an incorrect comment block.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Valgrind (memcheck and descriptor leaks check)

### CODE REVIEW

**Comment by palaciosjeremias** [comment]
### Valgrind report:
image

**Comment by palaciosjeremias** [comment]
### Scan-build report:
image

**Review by DProvinciani — CHANGES_REQUESTED** [review]
Good job! Just a couple of comments.

File: src/unit_tests/wazuh_db/test_wdb_global.c [hunk 1]
```diff
@@ -6732,6 +6732,609 @@ void test_wdb_global_unassign_agent_group_invalid_json(void **state) {
...
+    expect_string(__wrap__mdebug1, formatted_msg, "Error executing setting the agent group context: ERROR MESSAGE");
+
+    wdbc_result result = wdb_global_set_agent_group_context(data->wdb, agent_id, csv, hash, sync_status);
+
+    assert_int_equal(result, WDBC_ERROR);
+}
+
+/* wdb_global_set_agent_groups */
+
+void create_wdb_global_delete_agent_belong_success_call(int agent_id) {
```
[the remark this answers is not in the document]
Comment (palaciosjeremias) [reply src/unit_tests/wazuh_db/test_wdb_global.c:6819]: Changed

File: src/unit_tests/wazuh_db/test_wdb_global.c [hunk 2]
```diff
@@ -6732,6 +6732,609 @@ void test_wdb_global_unassign_agent_group_invalid_json(void **state) {
...
+    will_return(__wrap_wdb_begin2, 1);
+    will_return(__wrap_wdb_stmt_cache, 1);
+    expect_value(__wrap_sqlite3_bind_int, index, 1);
+    expect_value(__wrap_sqlite3_bind_int, value, agent_id);
+    will_return(__wrap_sqlite3_bind_int, SQLITE_OK);
+    will_return(__wrap_wdb_step, SQLITE_DONE);
+}
+
+void create_wdb_global_assign_agent_group_success_call(int agent_id, int group_id, char* group_name, cJSON* find_group_resp) {
```
[the remark this answers is not in the document]
Comment (palaciosjeremias) [reply src/unit_tests/wazuh_db/test_wdb_global.c:6835]: Changed

File: src/unit_tests/wazuh_db/test_wdb_global.c [hunk 3]
```diff
@@ -6732,6 +6732,609 @@ void test_wdb_global_unassign_agent_group_invalid_json(void **state) {
...
+    expect_value(__wrap_sqlite3_bind_int, index, 2);
+    expect_value(__wrap_sqlite3_bind_int, value, agent_id);
+    will_return(__wrap_sqlite3_bind_int, SQLITE_OK);
+    expect_value(__wrap_sqlite3_bind_int, index, 3);
+    expect_value(__wrap_sqlite3_bind_int, value, actual_priority);
+    will_return(__wrap_sqlite3_bind_int, SQLITE_OK);
+    will_return(__wrap_wdb_step, SQLITE_DONE);
+}
+
+void create_wdb_global_unassign_agent_group_success_call(int agent_id, int group_id, char* group_name, cJSON* find_group_resp) {
```
[the remark this answers is not in the document]
Comment (palaciosjeremias) [reply src/unit_tests/wazuh_db/test_wdb_global.c:6874]: Changed

File: src/unit_tests/wazuh_db/test_wdb_global.c [hunk 4]
```diff
@@ -6732,6 +6732,609 @@ void test_wdb_global_unassign_agent_group_invalid_json(void **state) {
...
+    expect_value(__wrap_sqlite3_bind_int, index, 1);
+    expect_value(__wrap_sqlite3_bind_int, value, group_id);
+    will_return(__wrap_sqlite3_bind_int, SQLITE_OK);
+    expect_value(__wrap_sqlite3_bind_int, index, 2);
+    expect_value(__wrap_sqlite3_bind_int, value, agent_id);
+    will_return(__wrap_sqlite3_bind_int, SQLITE_OK);
+    will_return(__wrap_wdb_exec_stmt_silent, OS_SUCCESS);
+}
+
+void create_wdb_global_calculate_agent_group_csv_success_call(int agent_id, cJSON* j_group_resp) {
```
[the remark this answers is not in the document]
Comment (palaciosjeremias) [reply src/unit_tests/wazuh_db/test_wdb_global.c:6900]: Changed

File: src/unit_tests/wazuh_db/test_wdb_global.c [hunk 5]
```diff
@@ -6732,6 +6732,609 @@ void test_wdb_global_unassign_agent_group_invalid_json(void **state) {
...
+    will_return(__wrap_wdb_begin2, 1);
+    will_return(__wrap_wdb_stmt_cache, OS_SUCCESS);
+    expect_value(__wrap_sqlite3_bind_int, index, 1);
+    expect_value(__wrap_sqlite3_bind_int, value, agent_id);
+    will_return(__wrap_sqlite3_bind_int, SQLITE_OK);
+    will_return(__wrap_wdb_exec_stmt_single_column, j_group_resp);
+    expect_function_call(__wrap_cJSON_Delete);
+}
+
+void create_wdb_global_set_agent_group_context_success_call(int agent_id, char* csv, char* hash, char* sync_status) {
```
[the remark this answers is not in the document]
Comment (palaciosjeremias) [reply src/unit_tests/wazuh_db/test_wdb_global.c:6917]: Changed

File: src/unit_tests/wazuh_db/test_wdb_global.c [hunk 6]
```diff
@@ -6732,6 +6732,609 @@ void test_wdb_global_unassign_agent_group_invalid_json(void **state) {
...
+    expect_value(__wrap_sqlite3_bind_text, pos, 3);
+    expect_string(__wrap_sqlite3_bind_text, buffer, sync_status);
+    will_return(__wrap_sqlite3_bind_text, SQLITE_OK);
+    expect_value(__wrap_sqlite3_bind_int, index, 4);
+    expect_value(__wrap_sqlite3_bind_int, value, agent_id);
+    will_return(__wrap_sqlite3_bind_int, SQLITE_OK);
+    will_return(__wrap_wdb_exec_stmt_silent, OS_SUCCESS);
+}
+
+void create_wdb_global_get_agent_max_group_priority_success_call(int agent_id, cJSON* j_priority_resp) {
```
[the remark this answers is not in the document]
Comment (palaciosjeremias) [reply src/unit_tests/wazuh_db/test_wdb_global.c:6940]: Changed

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**